### ISSUE TITLE
workingDirectoriesRoot ignored

### ISSUE THREAD

**Issue by hopfensp** [spring-cloud/spring-cloud-dataflow#3437]
When launching a task through the UI, the following property is ignored: deployer.app.local.workingDirectoriesRoot.  All log files are placed in the directory given by java.io.tmpdir regardless of the value that is set.

**Comment by sabbyanandan** [comment]
Hi, @hopfensp. Can you share the Task definition and the launch properties in entirety? Please make sure you share as much as information you can, so we can review and help you out.

**Comment by hopfensp** [comment]
I'm running locally on a RHEL 7 VM.  This is how I start cloud data flow server

[Unit]
Description=Spring Cloud Dataflow Server
After=syslog.target

[Service]
User=tomcat
ExecStart=/opt/java11/bin/java \
     -Xms256m \
     -Xmx256m \
     -DLOG_PATH=<path> \
     -jar /opt/brookdale/apps/spring-cloud-dataflow-server-2.2.0.RELEASE.jar \
     --spring.cloud.config.uri=<url> \
     --spring.application.name=scdf \
     --spring.profiles.active=test \
     --encrypt.key=<key> \
     --spring.cloud.dataflow.applicationProperties.task.spring.profiles.active=test \
     --spring.cloud.dataflow.applicationProperties.task.spring.cloud.config.uri=<url> \
     --spring.cloud.dataflow.applicationProperties.task.encrypt.key=<key>
SuccessExitStatus=143
TimeoutStartSec=infinity

[Install]
WantedBy=multi-user.target

Through the UI, I define an app called community, type is task and artifact is pulled from a nexus repo.

The task definition is very simple, start -> community -> end.

when I launch the task, in the parameters box, I specify deployer.community.local.workingDirectoriesRoot=<path> and the log files are dumped in /tmp.
I've tried defining inheritLogging and passing javaOpts, I have no issues with those, but only workingDirectoriesRoot.

**Comment by sabbyanandan** [comment]
Thank you for the update, @hopfensp. From the quick review, it does appear to be a bug. Both for streams and the tasks, it would exhibit the same behavior also. We will review it further and get back to you with an update.

**Comment by sabbyanandan** [comment]
@jvalkeal: I tried the following env-var variants in SCDF to globally override the root directory. Though the env-vars show up correctly at http://localhost:9393/management/env, I still see the logs ending up at `var/folders/..` (on MacOS).

```
spring.cloud.deployer.local.working-directories-root=file:/Users/sanandan/Downloads/fooz
spring_cloud_deployer_local_workingDirectoriesRoot=file:/Users/sanandan/Downloads/fooz
spring.cloud.deployer.local.workingDirectoriesRoot=file:/Users/sanandan/Downloads/fooz
```

I attempted to then override it at each stream/task launches, but that exhibits the same behavior. Logs end up at `var/folders/..`.

```
dataflow:>task create test --definition "timestamp"
dataflow:>task launch --name test --properties "deployer.*.local.working-directories-root=file:/Users/sanandan/Downloads/fooz"

dataflow:>stream create foo --definition "time | log"
dataflow:>stream deploy --name foo --properties "deployer.*.local.workingDirectoriesRoot=file:/Users/sanandan/Downloads/fooz"
```

**Comment by sabbyanandan** [comment]
Hi, @hopfensp: A quick update. 

With `spring.cloud.dataflow.task.platform.local.accounts.default.workingDirectoriesRoot=/Users/sanandan/Downloads/fooz` as the environment variable set at SCDF, every task launches and the stream deployments were successfully provisoned at the  `/Users/sanandan/Downloads/fooz` directory.

This is a global setting, however. That's why every deployment went to that pinned directory.

There seems to be an anomaly still with per-task/per-stream level deployment, and we will continue to troubleshoot it. In the meantime, you can try out the global configuration in your set up as a workaround for now.

**Comment by hopfensp** [comment]
Hi,

Thanks for your feedback.

I was able to figure out a temporary workaround for this by defining logging.file when I launch the task by passing app.<app>.logging.file.

I'll continue to do this and also define spring.cloud.dataflow.task.platform.local.accounts.default.workingDirectoriesRoot as well.

**Comment by markpollack** [comment]
It seems to be a bug.  There is a PR for it which we will review.  Glad you found a workaround.

https://github.com/spring-cloud/spring-cloud-deployer-local/pull/169

**Comment by markpollack** [comment]
Fixed in the deployer local.  Will be included in a backport for 2.2.x and included in 2.3.x releases.

https://github.com/spring-cloud/spring-cloud-deployer-local/commit/aac68f50816f3a194a75e34e96eae549140aafd3
